onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -71,7 +71,48 @@ void addOrtValueMethods(pybind11::module& m) {
 
         return ml_value;
       })
+      .def("update_inplace", [](OrtValue* ml_value, const py::array& py_values) {
+        if (!IsNumericNumpyArray(py_values)) {
+          throw std::runtime_error("Inplace update of OrtValues is currently only supported from non-string numpy arrays");
+        }
+
+        auto values_type = GetNumpyArrayType(py_values);
+        const auto device = ml_value->Get<Tensor>().Location().device;
+        if (device.Type() == OrtDevice::CPU) {
+          onnxruntime::python::CopyDataToTensor(
```

Comment:
Needs a check somewhere to ensure number of elements in py_values is the same as the size of the tensor buffer in ml_value ? Pretty sure CopyDataToTensor() doesn't have that check.

Reply:
Done